Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/keydownload/ui/KeyDownloadTestFragmentViewModel.kt

```diff
@@ -0,0 +1,75 @@
+package de.rki.coronawarnapp.test.keydownload.ui
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.diagnosiskeys.download.KeyPackageSyncSettings
+import de.rki.coronawarnapp.diagnosiskeys.download.KeyPackageSyncTool
+import de.rki.coronawarnapp.diagnosiskeys.storage.KeyCacheRepository
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.network.NetworkStateProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.sample
+import kotlinx.coroutines.runBlocking
+
+class KeyDownloadTestFragmentViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val syncSettings: KeyPackageSyncSettings,
+    networkStateProvider: NetworkStateProvider,
+    private val keyPackageSyncTool: KeyPackageSyncTool,
+    private val keyCacheRepository: KeyCacheRepository
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    val currentCache = runBlocking {
+        // TODO runBlocking is not nice, how can we solve this better?
```

Comment:
It's only consumed by the fragment as far as I can see. Why not make it a suspending function and call it lifecycle-aware from the fragment?